src/auth.ts

```diff
@@ -34,4 +78,10 @@ export default class Auth extends AinftBase {
     const trailingUrl = `user/${userId}/ethAddress`;
     return this.sendRequest(HttpMethod.DELETE, trailingUrl, query);
   }
+
+  private async getInitialStakeAmount(appId: string, userId: string) {
+    const user = await this.getUser(appId, userId);
+    const balance = await this.ain.wallet.getBalance(user.address);
+    return balance - INITIALIZE_GAS_FEE;
```

Comment:
can this amount calculation logic moved to the ainft server side?
(maybe a todo here at this time?)

Reply:
If this logic is moved to the server side, the stake API will always stake all the user's balance if the amount does not come in, wouldn't it be dangerous?

Reply:
@woomurf what do you mean by 'the amount does not come in'?

The ain account that does the staking is created and managed not by the user but ainft server, and the balance will be the amount transferred from the pool by the ainft server as well. So there will no no concerns on the balance?

Reply:
When using the stake API, if the user stakes all of the user's balance, isn't the amount argument value unnecessary?

I think that the user can additionally stake in the app in the future, and for this, the user should get the address of their nft server account and transfer there and then staking using stake API.